### PR TITLE
[Google Analytics] GA toggle on/off

### LINKED ISSUE #8190: toggle to turn off GA
# Story

As a web team, we would like the ability to quickly toggle off Google Analytics for a site in case it is required because of HIPPA or other reasons.

# Additional Info

Initial slice related to the Epic https://github.com/uiowa/uiowa/issues/8180

Discussed adding a toggle to turn off GA similar to the toggle for GTM. Should output a message on the GA config page saying that this is turned off and to contact our team. Talked about moving forward with this for now and continue learning and working towards giving site owners more control to take this responsibility on in future slices.
```[tasklist]
### Tasks
- [x] Look for a contrib patch could allow them to just remove their property id to remove tracking
- [x] Add toggle to turn on/off GA on uiowa_core settings page
```

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uiowa/issues/8190. 

# How to test
1. Code review
2. Comment out these lines locally https://github.com/uiowa/uiowa/blob/main/docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php#L19-L25. 

**Sandbox**
```
ddev composer install && ddev blt ds --site=sandbox.uiowa.edu &&  ddev drush @sandbox.local uli /admin/config/sitenow/uiowa-core
```
**CCOM**
```
 ddev composer install && ddev blt ds --site=dermatology.medicine.uiowa.edu &&  ddev drush @medicinedermatology.local uli /admin/config/sitenow/uiowa-core
```
3. Enable/disable checkbox 
4. Go to admin/config/services/google-analytics
5. Try to remove IDs. 
6. Confirm GA form is disabled and there is a message at the top of the page.

### CODE REVIEW

**Comment by joewhitsitt** [comment]
pulling to test

File: docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php
Comment (joewhitsitt) [inline docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: I think this->configFactory is always non-NULL based on you bringing it in via dependency injection. Instead it should check that the config for uiowa_core exists.

I think this could be simplified to:
```php
    $uiowa_core_config = $this->configFactory->get('uiowa_core.settings');
    if ($uiowa_core_config && !$uiowa_core_config->get('uiowa_core.ga')) {
      $overrides['google_analytics.settings']['account'] = '';
    }
```

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: Care if I make this change @bspeare ?

Comment (bspeare) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: Please do! Thanks!

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: Actually, I am getting TypeErrors elsewhere related to this. Poking.

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: i think there is an issue with updating the services file like that (replacing request_stack). hard to tell if that will be an issue on deployment, but it appears to be an issue on syncing locally. We could add a `drupal_flush_caches` to the update hook but it may error before that.

```
TypeError: Drupal\sitenow\ConfigOverride\GoogleAnalyticsOve  
  rride::__construct(): Argument #1 ($config_factory) must be  
   of type Drupal\Core\Config\ConfigFactoryInterface, Symfony  
  \Component\HttpFoundation\RequestStack given, called in /va  
  r/www/html/docroot/core/lib/Drupal/Component/DependencyInje  
  ction/Container.php on line 261 in Drupal\sitenow\ConfigOve  
  rride\GoogleAnalyticsOverride->__construct() (line 28 of /v  
  ar/www/html/docroot/profiles/custom/sitenow/src/ConfigOverr  
  ide/GoogleAnalyticsOverride.php).  
```

Comment (bspeare) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: What site did you get that on?

Comment (bspeare) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: I had originally added the `if ($this->configFactory)` to get the test to pass that just failed on this build.

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: Any site I previously had pulled down and did a `blt ds` on. (e.g. sandbox, admissions)

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php]: steps to reproduce:

- checkout main
- sql-drop, cr, sql-sync
- uli. hit pages.
- checkout ga_delete
- hit pages (WSOD)
- updb
- hit pages (good?)

**Review by joewhitsitt — APPROVED** [review]
approving but if folks are uncomfortable with my tweaks let me know. main change is configFactory shouldn't be handled as optional and because the service files changes, there could be temporary error screens so a cache clear to try and speed things up.

alternatively, when this is deployed we could immediately follow with a `ume cr`

**Comment by bspeare** [comment]
Thanks @joewhitsitt!

**Comment by joewhitsitt** [comment]
ack! ~The test(s) are still~ Another test is failing. my bad

**Comment by cory-skeers** [comment]
> approving but if folks are uncomfortable with my tweaks let me know. main change is configFactory shouldn't be handled as optional and because the service files changes, there could be temporary error screens so a cache clear to try and speed things up.
> 
> alternatively, when this is deployed we could immediately follow with a `ume cr`

I'm good with your changes, so second the approve!

Do we want to do an off-hours deployment with this one? Leave it to the update hook to clear the caches automagically, but off-hours to limit the chance of disruption even if it's a smaller window?

**Pull request merged**
